model/lifecycle.go

```diff
@@ -418,29 +425,56 @@ func RestartBuild(buildId string, taskIds []string, abortInProgress bool, caller
 			return errors.WithStack(err)
 		}
 	}
+	// restart all the 'not in-progress' tasks for the build
+	allTasks, err := task.FindWithDisplayTasks(task.ByIdsBuildAndStatus(taskIds, buildId, task.CompletedStatuses))
```

Comment:
do we need to add display tasks' execution tasks? (like we do in `RestartVersion`)

Reply:
Update: I'm removing the refactor. I think I realized why resetTask isn't being used; it looks like we want to Archive only non-aborted tasks 🤔  I'm not 100% sure why the logic works this way but I don't want to severely break something with an unnecessary refactor so i'm going to leave it as is.